doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,291 @@
+
+The following SQL statement shows how to call an AAF query handler:
+
+```sql
+EXECUTE SCRIPT AAF_RUN_QUERY_HANDLER('{
+    "query_handler": {
+        "factory_class": {
+            "module": "<CLASS_MODULE>",
+            "name": "<CLASS_NAME>"
+        },
+        "parameters": "<CLASS_PARAMETERS>",
+        "udf": {
+            "schema": "<UDF_DB_SCHEMA>",
+            "name": "<UDF_NAME>"
+        }
+    },
+    "temporary_output": {
+        "bucketfs_location": {
+            "connection_name": "<BUCKETFS_CONNECTION_NAME>",
+            "directory": "<BUCKETFS_DIRECTORY>"
+        },
+        "schema_name": "<TEMP_DB_SCHEMA>"
+    }
+}');
+```
+
+See [Implementation of Custom Algorithms](#implementation-of-custom-algorithms) for a complete example.
+
+### Parameters
+
+| Parameter                    | Required? | Description                                                                                                    |
+|------------------------------|-----------|----------------------------------------------------------------------------------------------------------------|
+| `<CLASS_NAME>`               | yes       | Name of the query handler class                                                                                |
+| `<CLASS_MODULE>`             | yes       | Module name of the query handler class <span style="color: red">(should we mention `builtins` here?)</span>    |
+| `<CLASS_PARAMETERS>`         | yes       | Parameters of the query handler class encoded as string                                                        |
+| `<UDF_NAME>`                 | -         | Name of Python UDF script including user-implemented algorithm                                                 |
+| `<UDF_DB_SCHEMA>`            | -         | Schema name where the UDF script is deployed                                                                   |
+| `<BUCKETFS_CONNECTION_NAME>` | yes       | BucketFS connection name to create temporary file outputs                                                      |
+| `<BUCKETFS_DIRECTORY>`       | yes       | Directory in BucketFS for temporary file outputs                                                               |
+| `<TEMP_DB_SCHEMA>`           | yes       | Database Schema for temporary database objects, e.g. tables                                                    |
+
+# Implementation of Custom Algorithms
+
+Each algorithm should extend the `UDFQueryHandler` abstract class and then implement the following methods:
+* `start()`: This method is called at the first execution of the framework, that is, in the first iteration. It returns a result object: Either _Finish_ or _Continue_.
+  * The _Finish_ result object contains the final result of the implemented algorithm.
+  * The _Continue_ object contains the query list that will be executed for the next state.
+* `handle_query_result()`: This method is called at the following iterations to handle the return query.
+
```

Comment:
We probably need to mention usage modes for the AAF,

1) Adhoc implementation of algorithms within a UDF
2) Building an extension
  - means python package which implements the query handler and an associated SLC which has the python package installed 
  - here we only give a link to PEC
  - we give an example how would need to AAF_RUN_QUERY_HANDLER in this case

Reply:
This sentence I don't understand. 

> we give an example how would need to AAF_RUN_QUERY_HANDLER in this case

Reply:
> This sentence I don't understand.
> 
> > we give an example how would need to AAF_RUN_QUERY_HANDLER in this case

we give an example how would a user need to call AAF_RUN_QUERY_HANDLER in this case

Reply:
Thanks - see next push.